Crash (Delete a collection -> Tap Search field)

1. Launch Scryer
2. Create a collection
3. Delete the collection
4. Tap Search field

Reproduced Rate: 10/10

Scryer 1.0(191).nightly
Device:
 1. Vivo 1606 with Android 6.0.1
 2. Pixel 2 with Android 8.1.0

--------- beginning of crash
09-06 18:29:07.261 14611 14611 E AndroidRuntime: FATAL EXCEPTION: main
09-06 18:29:07.261 14611 14611 E AndroidRuntime: Process: gro.allizom.scryer, PID: 14611
09-06 18:29:07.261 14611 14611 E AndroidRuntime: java.lang.IllegalArgumentException: navigation destination gro.allizom.scryer:id/action_navigate_to_search is unknown to this NavController
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at androidx.navigation.NavController.navigate(NavController.java:671)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at androidx.navigation.NavController.navigate(NavController.java:630)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at org.mozilla.scryer.landingpage.HomeFragment$createOptionsMenuSearchView$4.onClick(HomeFragment.kt:376)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.view.View.performClick(View.java:6597)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.view.View.performClickInternal(View.java:6574)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.view.View.access$3100(View.java:778)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.view.View$PerformClick.run(View.java:25885)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:873)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:99)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:193)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6669)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
09-06 18:29:07.261 14611 14611 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
09-06 18:29:07.264  1187 12290 W ActivityManager:   Force finishing activity gro.allizom.scryer/org.mozilla.scryer.MainActivity
09-06 18:29:07.282 14611 14611 I Process : Sending signal. PID: 14611 SIG: 9
09-06 18:29:07.306  1187  1381 W InputDispatcher: channel '1d7ed00 gro.allizom.scryer (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
09-06 18:29:07.306  1187  1381 E InputDispatcher: channel '1d7ed00 gro.allizom.scryer (server)' ~ Channel is unrecoverably broken and will be disposed!
09-06 18:29:07.331  1187  1381 W InputDispatcher: channel 'd02820b gro.allizom.scryer (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
09-06 18:29:07.331  1187  1381 E InputDispatcher: channel 'd02820b gro.allizom.scryer (server)' ~ Channel is unrecoverably broken and will be disposed!
09-06 18:29:07.331  1187  1381 W InputDispatcher: channel 'ccd38fc gro.allizom.scryer/org.mozilla.scryer.MainActivity (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
09-06 18:29:07.331  1187  1381 E InputDispatcher: channel 'ccd38fc gro.allizom.scryer/org.mozilla.scryer.MainActivity (server)' ~ Channel is unrecoverably broken and will be disposed!
09-06 18:29:07.331  1187  2169 I ActivityManager: Process gro.allizom.scryer (pid 14611) has died: vis  +99TOP 
09-06 18:29:07.332  1187  1201 I WindowManager: WIN DEATH: Window{1d7ed00 u0 gro.allizom.scryer}
09-06 18:29:07.332  1187  1201 W InputDispatcher: Attempted to unregister already unregistered input channel '1d7ed00 gro.allizom.scryer (server)'
09-06 18:29:07.332  1187  2169 W ActivityManager: Scheduling restart of crashed service gro.allizom.scryer/org.mozilla.scryer.ScryerService in 1000ms
09-06 18:29:07.332  1187  1211 W libprocessgroup: kill(-14611, 9) failed: No such process
09-06 18:29:07.332  1187  1211 I libprocessgroup: Successfully killed process cgroup uid 10198 pid 14611 in 0ms
09-06 18:29:07.333   712   712 I Zygote  : Process 14611 exited due to signal (9)
09-06 18:29:07.333   614  1391 W SurfaceFlinger: Attempting to destroy on removed layer: 1d7ed00 gro.allizom.scryer#0
09-06 18:29:07.335  1187 10168 I WindowManager: WIN DEATH: Window{d02820b u0 gro.allizom.scryer}
09-06 18:29:07.335  1187 10168 W InputDispatcher: Attempted to unregister already unregistered input channel 'd02820b gro.allizom.scryer (server)'
09-06 18:29:07.337   614  1391 W SurfaceFlinger: Attempting to destroy on removed layer: d02820b gro.allizom.scryer#0
09-06 18:29:07.337  1187  4627 I WindowManager: WIN DEATH: Window{ccd38fc u0 gro.allizom.scryer/org.mozilla.scryer.MainActivity}
09-06 18:29:07.337  1187  4627 W InputDispatcher: Attempted to unregister already unregistered input channel 'ccd38fc gro.allizom.scryer/org.mozilla.scryer.MainActivity (server)'
09-06 18:29:07.341   614  1391 W SurfaceFlinger: Attempting to set client state on removed layer: gro.allizom.scryer/org.mozilla.scryer.MainActivity#0
09-06 18:29:07.341   614  1391 W SurfaceFlinger: Attempting to destroy on removed layer: gro.allizom.scryer/org.mozilla.scryer.MainActivity#0
09-06 18:29:07.344  1647  1647 W StaticLayout: maxLineHeight should not be -1.  maxLines:1 lineCount:1
09-06 18:29:07.344  1647  1647 W StaticLayout: maxLineHeight should not be -1.  maxLines:1 lineCount:1
09-06 18:29:07.354  1187  1216 W ActivityManager: setHasOverlayUi called on unknown pid: 14611
09-06 18:29:07.354  1187  1216 I chatty  : uid=1000(system) android.display identical 1 line
09-06 18:29:07.354  1187  1216 W ActivityManager: setHasOverlayUi called on unknown pid: 14611
09-06 18:29:07.387  6407  6407 I RunnerModule: Event Bus thread is disabled
09-06 18:29:07.388  6407  6407 I SearchServiceCore: Event Bus thread is disabled; marking main thread as EventBus
09-06 18:29:07.394  6407  6407 W SessionLifecycleManager: Handover failed. Creating new session controller.
09-06 18:29:07.400  6407 14415 I RunnerModule: Event Bus thread is disabled
09-06 18:29:07.402  1755  1755 I WallpaperService: engine resumed
09-06 18:29:07.409   827   894 D CHRE    : @ 96546.961: +: id 10, otherClientPresent 1, mode 3
09-06 18:29:07.410   827   894 D CHRE    : @ 96546.969: [ImuCal] Dynamic sensor configuration: high-performance.
09-06 18:29:07.410  6407  6407 I StreamController: registerScope: nowcards--3827724578765767544
09-06 18:29:07.410   827   894 D CHRE    : @ 96546.969: sensorType 9 allowed 1: mergedMode 1, otherClientPresent 1
09-06 18:29:07.411   827   894 D CHRE    : @ 96546.969: sensorType 12 allowed 1: mergedMode 1, otherClientPresent 1
09-06 18:29:07.412   827   894 D CHRE    : @ 96546.969: sensorType 14 allowed 0: mergedMode 3, otherClientPresent 0
09-06 18:29:07.427  6407  6407 W TemplatedStringEvaluato: Got templated string with no display string
09-06 18:29:07.428  6407  6407 I ClusterController: Found empty cluster, or no children successfully created.
09-06 18:29:07.437  6407 14423 W LocationOracle: No location history returned by ContextManager
09-06 18:29:07.448  6407  6407 I StreamRenderer: maybeRegisterNowCardsScope(nowcards--3827724578765767544, com.google.android.sidekick.shared.remoteapi.CardRenderingContext@5e8b66a)
09-06 18:29:07.477  2609  4138 I Places  : ?: PlacesBleScanner start() with priority 2
09-06 18:29:07.481  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.482  6407  6407 I chatty  : uid=10047 com.google.android.googlequicksearchbox:search identical 2 lines
09-06 18:29:07.482  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.483  2609  4138 I PlaceInferenceEngine: [anon] Changed inference mode: 1
09-06 18:29:07.484  2609  4138 I PlaceInferenceEngine: [account#-1795736791] Changed inference mode: 1
09-06 18:29:07.486  2609  4138 I Places  : Converted 1 out of 46 WiFi scans
09-06 18:29:07.490  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:3 lineCount:3
09-06 18:29:07.490  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.491  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:3 lineCount:3
09-06 18:29:07.491  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.491  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:3 lineCount:3
09-06 18:29:07.492  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.492  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:3 lineCount:3
09-06 18:29:07.492  6407  6407 W StaticLayout: maxLineHeight should not be -1.  maxLines:2 lineCount:2
09-06 18:29:07.495  2609 15097 I PlaceInferenceEngine: No beacon scan available - ignoring candidates.
09-06 18:29:07.506  4126  4186 I PlayCommon: [571] com.google.android.play.b.g.d(216): Preparing logs for uploading
09-06 18:29:07.513  6407  6407 W MonetServiceEventQueue: No updates to flush
09-06 18:29:07.520  4126  4186 I PlayCommon: [571] com.google.android.play.b.g.a(385): Connecting to server: https://play.googleapis.com/play/log?format=raw&proto_v2=true
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: Failure parsing personalized places proto.
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: bkwg: While parsing a protocol message, the input ended unexpectedly in the middle of a field.  This could mean either that the input has been truncated or that an embedded message misreported its own length.
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at bkvo.a(:com.google.android.gms@12874041@12.8.74 (100400-204998136):137)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at bkvo.b(:com.google.android.gms@12874041@12.8.74 (100400-204998136):19)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at azbx.a(:com.google.android.gms@12874041@12.8.74 (100400-204998136):13)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at qbv.run(:com.google.android.gms@12874041@12.8.74 (100400-204998136):3)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at android.os.Handler.handleCallback(Handler.java:873)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at android.os.Handler.dispatchMessage(Handler.java:99)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at android.os.Looper.loop(Looper.java:193)
09-06 18:29:07.526  2609  4138 E PlaceInferenceEngine: 	at android.os.HandlerThread.run(HandlerThread.java:65)
09-06 18:29:07.547  2609  4138 I Places  : ?: PlacesBleScanner start() with priority 2
09-06 18:29:07.554  2609  4138 I Places  : ?: PlacesBleScanner start() with priority 2
09-06 18:29:07.554  2609  4138 I PlaceInferenceEngine: [anon] Changed inference mode: 1
09-06 18:29:07.554  2609  4138 I PlaceInferenceEngine: [account#-1795736791] Changed inference mode: 1
09-06 18:29:07.568  2609  4138 I Places  : Converted 1 out of 46 WiFi scans
09-06 18:29:07.603  4126  4186 I PlayCommon: [571] com.google.android.play.b.g.a(451): Successfully uploaded logs.
09-06 18:29:07.606  2609 15097 I PlaceInferenceEngine: No beacon scan available - ignoring candidates.
09-06 18:29:07.659  2609  4079 E Volley  : [197] BasicNetwork.performRequest: Unexpected response code 403 for https://www.googleapis.com/placesandroid/v1/getPlaceById?key=AIzaSyAf4nrRiEKvqzlRKTncQaAXMzb3ePYHr8Y
09-06 18:29:07.663  2609 15177 W Places  : {"code":403,"errors":[{"reason":"forbidden","domain":"global","message":"Requests from this Android client application com.google.android.apps.nexuslauncher are blocked."}]}
09-06 18:29:07.668  2609 15177 E AsyncOperation: serviceID=65, operation=GetPlaceById
09-06 18:29:07.668  2609 15177 E AsyncOperation: OperationException[Status{statusCode=ERROR, resolution=null}]
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at azjg.b(:com.google.android.gms@12874041@12.8.74 (100400-204998136):1)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at aziz.a(:com.google.android.gms@12874041@12.8.74 (100400-204998136):27)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at vyd.run(:com.google.android.gms@12874041@12.8.74 (100400-204998136):33)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at bbwe.run(:com.google.android.gms@12874041@12.8.74 (100400-204998136):2)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at pis.run(:com.google.android.gms@12874041@12.8.74 (100400-204998136):28)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at pom.run(Unknown Source:7)
09-06 18:29:07.668  2609 15177 E AsyncOperation: 	at java.lang.Thread.run(Thread.java:764)
09-06 18:29:07.694  2609 15097 I PlaceInferenceEngine: No beacon scan available - ignoring candidates.
09-06 18:29:07.697  2609 15177 W ctxmgr  : [AclManager]No 2 for (accnt=account#-1795736791#, com.google.android.gms(10021):PlacesProducer, vrsn=12874000, 0, 3pPkg = null ,  3pMdlId = null ,  pid = 2609). Was: 2 for 18, account#-1795736791#
09-06 18:29:07.719  2609 15248 I NetRec  : [2821] aesa.a: Refreshing scores for 2 networks.
09-06 18:29:07.746  2609  4078 W Conscrypt: Could not set socket write timeout: java.lang.reflect.InvocationTargetException
09-06 18:29:07.747  2609  4078 W Conscrypt: 	at java.lang.reflect.Method.invoke(Native Method)
09-06 18:29:07.747  2609  4078 W Conscrypt: 	at com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@12874041@12.8.74 (100400-204998136):13)
09-06 18:29:07.747  2609  4078 W Conscrypt: Caused by: android.system.ErrnoException: setsockopt failed: EBADF (Bad file descriptor)
09-06 18:29:07.747  2609  4078 W Conscrypt: 	at libcore.io.Linux.setsockoptTimeval(Native Method)
09-06 18:29:07.747  2609  4078 W Conscrypt: 	at libcore.io.ForwardingOs.setsockoptTimeval(ForwardingOs.java:174)
09-06 18:29:07.764  2609 15097 I PlaceInferenceEngine: No beacon scan available - ignoring candidates.
09-06 18:29:07.775  2609  4078 W Conscrypt: Could not set socket write timeout: java.lang.reflect.InvocationTargetException
09-06 18:29:07.776  2609  4078 W Conscrypt: 	at java.lang.reflect.Method.invoke(Native Method)
09-06 18:29:07.776  2609  4078 W Conscrypt: 	at com.google.android.gms.org.conscrypt.Platform.setSocketWriteTimeout(:com.google.android.gms@12874041@12.8.74 (100400-204998136):13)
09-06 18:29:07.776  2609  4078 W Conscrypt: Caused by: android.system.ErrnoException: setsockopt failed: EBADF (Bad file descriptor)
09-06 18:29:07.776  2609  4078 W Conscrypt: 	at libcore.io.Linux.setsockoptTimeval(Native Method)
09-06 18:29:07.776  2609  4078 W Conscrypt: 	at libcore.io.ForwardingOs.setsockoptTimeval(ForwardingOs.java:174)
09-06 18:29:07.873  2609 15248 I NetRec  : [2821] aesa.a: Completed refresh_scores_task score refresh task in 158 ms, returning 0
09-06 18:29:08.361  1187  1210 I ActivityManager: Start proc 15250:gro.allizom.scryer/u0a198 for service gro.allizom.scryer/org.mozilla.scryer.ScryerService
09-06 18:29:08.385 15250 15250 E .allizom.scrye: Not starting debugger since process cannot load the jdwp agent.
09-06 18:29:08.416 15250 15250 I .allizom.scrye: The ClassLoaderContext is a special shared library.
09-06 18:29:08.450 15250 15250 D FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
09-06 18:29:08.456 15250 15250 D FirebaseApp: com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
09-06 18:29:08.456 15250 15250 I FirebaseInitProvider: FirebaseApp initialization successful
09-06 18:29:08.486 15250 15268 I FA      : App measurement is starting up, version: 12780
09-06 18:29:08.486 15250 15268 I FA      : To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
09-06 18:29:08.486 15250 15268 I FA      : To enable faster debug mode event logging run:
09-06 18:29:08.486 15250 15268 I FA      :   adb shell setprop debug.firebase.analytics.app gro.allizom.scryer
09-06 18:29:08.500  1187  1187 I NotificationService: Cannot find enqueued record for key: 0|gro.allizom.scryer|9487|null|10198
09-06 18:29:10.308   827   894 I VSC     : @ 96549.859: [VSC] Request sensor 131074, sample_duration 0 ms, report_duration 0 ms
09-06 18:29:10.308   827   894 I VSC     : @ 96549.867: [VSC] Stopping sensor events for sensor 65537
09-06 18:29:10.311   827   894 I CHRE    : @ 96549.867: [AR_CHRE] still: 66
09-06 18:29:10.311   827   894 I CHRE    : @ 96549.867: [AR_CHRE] ON => IDLE
09-06 18:29:11.252   827   894 I VSC     : @ 96550.805: [VSC] Stopping sensor events for sensor 131074
09-06 18:29:11.252   827   894 I CHRE    : @ 96550.805: [AR_CHRE] IDLE => ON
09-06 18:29:11.252   827   894 I VSC     : @ 96550.805: [VSC] Request sensor 65537, sample_duration 14 ms, report_duration c80 ms
09-06 18:29:12.042  1187  1187 V SettingsProvider: Notifying for 0: content://settings/system/radio.data.stall.recovery.action
09-06 18:29:12.045 13455 15242 I Authzen : [DeviceStateSyncManager] The server is in sync with current state. Nothing to do
09-06 18:29:12.052  1187  1415 D ConnectivityService: NetworkAgentInfo [MOBILE (LTE) - 172] EVENT_NETWORK_INFO_CHANGED, going from CONNECTED to DISCONNECTED
09-06 18:29:12.053  1187  1415 D ConnectivityService: NetworkAgentInfo [MOBILE (LTE) - 172] got DISCONNECTED, was satisfying 16
09-06 18:29:12.053  1855  1984 D DcNetworkAgent: NetworkAgent: NetworkAgent channel lost
09-06 18:29:12.053  1855  1984 D DcNetworkAgent: NetworkAgent: DcNetworkAgent: unwanted found mNetworkAgent=null, which isn't me.  Aborting unwanted
09-06 18:29:12.068  1816  1909 I QCNEJ/CndHalConnector: -> SND notifyWwanStatusChanged(RatInfo{type=0, subType=13, state=CONNECTED, v4=IfaceAddrPair{iface=rmnet_data0, addr=10.161.223.170}, v6=IfaceAddrPair{iface=, addr=}, timestamp=2018-09-06 18:29:12.067, isAndroidValidated=false, netHandle=742140071949, slotIdx=0} CellularInfo{signalStrength=-1, isRoaming=false})
09-06 18:29:12.068  1816  1909 I QCNEJ/CndHalConnector: -> SND notifyRatConnectionStatusChanged(rat = 0, CONNECTED, v4 = 10.161.223.170, v6 = )
09-06 18:29:12.074  1816  1907 I QCNEJ/CndHalConnector: -> SND notifyWwanStatusChanged(RatInfo{type=0, subType=13, state=CONNECTED, v4=IfaceAddrPair{iface=rmnet_data0, addr=10.161.223.170}, v6=IfaceAddrPair{iface=, addr=}, timestamp=2018-09-06 18:29:12.073, isAndroidValidated=false, netHandle=742140071949, slotIdx=0} CellularInfo{signalStrength=-1, isRoaming=false})
09-06 18:29:12.091   699   699 D TcpSocketMonitor: resuming tcpinfo polling (interval=30000ms)
09-06 18:29:12.125  1187  1214 I EthernetTracker: interfaceLinkStateChanged, iface: rmnet_data0, up: false
09-06 18:29:12.183   699   716 E Netd    : Interface index 9 was never in a currently-connected netId
09-06 18:29:12.183   699 14294 W DnsTlsSocket: SSL_connect error 5, errno=103
09-06 18:29:12.184   699 14294 W ResolverController: netId 172 was erased during private DNS validation
09-06 18:29:12.184   699   716 I Netd    : Destroyed 2 sockets on 10.161.223.170 in 0.7 ms
09-06 18:29:12.184   699 14295 W DnsTlsSocket: SSL_connect error 5, errno=103
09-06 18:29:12.184   699 14295 W ResolverController: netId 172 was erased during private DNS validation
09-06 18:29:12.466  6407 14504 I EventLogSendingHelper: Sending log events.
09-06 18:29:13.044  1187 15119 D NetworkMonitor/NetworkAgentInfo [MOBILE (LTE) - 172]: PROBE_HTTPS https://www.google.com/generate_204 Probe failed with exception java.net.UnknownHostException: www.google.com: TLS-bypass resolution failed
09-06 18:29:13.855  1187  1187 V SettingsProvider: Notifying for 0: content://settings/system/screen_brightness

--- a/app/src/main/java/org/mozilla/scryer/landingpage/CollectionFragment.kt
+++ b/app/src/main/java/org/mozilla/scryer/landingpage/CollectionFragment.kt
@@ -135,7 +135,12 @@ class CollectionFragment : Fragment() {
 
             R.id.action_collection_delete -> {
                 context?.let {
-                    showDeleteCollectionDialog(view, ScreenshotViewModel.get(this), collectionId)
+                    showDeleteCollectionDialog(it, ScreenshotViewModel.get(this), collectionId,
+                            object : OnDeleteCollectionListener {
+                                override fun onDeleteCollection() {
+                                    Navigation.findNavController(view).navigateUp()
+                                }
+                            })
                 }
             }
             else -> return super.onOptionsItemSelected(item)
@@ -341,6 +346,10 @@ interface OnContextMenuActionListener {
     fun onContextMenuAction(item: MenuItem?, itemPosition: Int)
 }
 
+interface OnDeleteCollectionListener {
+    fun onDeleteCollection()
+}
+
 fun showScreenshotInfoDialog(context: Context, screenshotModel: ScreenshotModel) {
     val message = StringBuilder()
     message.apply {
@@ -477,8 +486,7 @@ private fun showCollectionInfoDialog(context: Context, collection: CollectionMod
             .show()
 }
 
-fun showDeleteCollectionDialog(view: View, viewModel: ScreenshotViewModel, collectionId: String?) {
-    val context = view.context
+fun showDeleteCollectionDialog(context: Context, viewModel: ScreenshotViewModel, collectionId: String?, listener: OnDeleteCollectionListener?) {
     AlertDialog.Builder(context)
             .setTitle(context.getString(R.string.dialogue_deletecollection_title_delete))
             .setMessage(context.getString(R.string.dialogue_delete_content_cantundo))
@@ -495,7 +503,7 @@ fun showDeleteCollectionDialog(view: View, viewModel: ScreenshotViewModel, colle
                     }
                 }
                 dialog?.dismiss()
-                Navigation.findNavController(view).navigateUp()
+                listener?.onDeleteCollection()
             }
             .show()
 }

--- a/app/src/main/java/org/mozilla/scryer/landingpage/MainAdapter.kt
+++ b/app/src/main/java/org/mozilla/scryer/landingpage/MainAdapter.kt
@@ -110,7 +110,7 @@ class MainAdapter(private val fragment: Fragment?): RecyclerView.Adapter<Recycle
         when (item?.itemId) {
             CONTEXT_MENU_ID_RENAME -> fragment?.context?.let { CollectionNameDialog.renameCollection(it, ScreenshotViewModel.get(fragment), collectionModel.id) }
             CONTEXT_MENU_ID_INFO -> fragment?.context?.let { showCollectionInfo(it, ScreenshotViewModel.get(fragment), collectionModel.id) }
-            CONTEXT_MENU_ID_DELETE -> fragment?.context?.let { showDeleteCollectionDialog(fragment.view!!, ScreenshotViewModel.get(fragment), collectionModel.id) }
+            CONTEXT_MENU_ID_DELETE -> fragment?.context?.let { showDeleteCollectionDialog(it, ScreenshotViewModel.get(fragment), collectionModel.id, null) }
         }
     }
 

